fix(FE): rendering order of values on the log details page

mustajab-ikram opened the pull request:
fix(FE): Ensure proper order of values on logs detail page

![image](https://github.com/SigNoz/signoz/assets/94978425/c8b431f9-9343-4435-a680-cbb49af09a2a)

closes issue #3158 . The previous version had a specific ordering for timestamp, id, trace_id, span_id, trace_flags, severity_number, severity_text, body, attributes_string.container_id, attributes_string.log_file_path, and attributes_string.stream on the logs detail page. The new detail page was rendering the values in random order. This commit updates the rendering logic to respect the defined fieldOrder array, ensuring the values are displayed in the correct order.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SigNoz/signoz?pullRequest=3189) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SigNoz/signoz?pullRequest=3189) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SigNoz/signoz?pullRequest=3189) it.</sub>

palashgdev commented on frontend/src/container/LogDetailedView/TableView.tsx:
```diff
@@ -51,10 +51,33 @@ function TableView({
 		flattenLogData !== null &&
 		Object.keys(flattenLogData)
 			.filter((field) => fieldSearchFilter(field, fieldSearchInput))
+			.sort((a, b) => {
```
can we explicitly make the timestamp and id first ... and rest server controls ?

mustajab-ikram replied:
done

palashgdev commented:
also let's make a hook / util so that it is same across every view for same log data

mustajab-ikram commented on frontend/src/container/LogDetailedView/TableView.tsx:
```diff
@@ -51,10 +51,33 @@ function TableView({
 		return null;
 	}
 
+	const fieldOrder = [
```
this is addressed in next review comment

mustajab-ikram commented:
This commit introduces a separate hook, 'useLogData', in the 'utils.ts' file to handle sorting and processing of log data in the 'TableView' component. Previously, the sorting logic was directly present in the 'TableView' component.

The hook takes care of sorting the log data fields in a specific order as per your suggestion. It ensures that the fields are displayed in the correct order on the logs detail page.

Changes Made:
- Moved sorting logic and data processing from 'TableView' to the 'useLogData' hook in 'utils.ts'.
- Replaced direct sorting logic in 'TableView' with the 'sortDataByField' function from the hook.

Reviewed and addressed the feedback on the initial implementation. 
> also let's make a hook / util so that it is same across every view for same log data

palashgdev commented on frontend/src/container/LogDetailedView/utils.ts:
```diff
@@ -32,3 +36,42 @@ export function flattenObject(obj: AnyObject, prefix = ''): AnyObject {
+	| { key: string; field: string; value: string }[]
+	| null;
+
+export function useLogData(logData: ILog): UseLogDataReturnType {
```
this is util can we make this getSortedLogDetailsData

nityanandagohain commented:
Something is wrong, the changes doesn't seem to be working.

Also please use proper branch names don't add `/` in branch names.

mustajab-ikram commented:
> Something is wrong, the changes doesn't seem to be working.
> 
> Also please use proper branch names don't add `/` in branch names.

hey @nityanandagohain,  I thoroughly tested the changes on my end before submitting them. Could you please provide me with more details about the specific problems you encountered? Any additional information, such as any unexpected behavior you encountered, will be helpful in addressing the issue promptly.

nityanandagohain commented:
NVM, it was a branch issue, since it had '/' the changes were not added.

nityanandagohain reviewed (commented):
I see that the ordering is a bit differet, this is what I see 
<img width="1054" alt="image" src="https://github.com/SigNoz/signoz/assets/26831659/2e8a5449-8c00-46ae-9b31-815915659051">


What we we want is like this 
<img width="1054" alt="image" src="https://github.com/SigNoz/signoz/assets/26831659/5fa3a141-a1fd-42b1-a98f-9ff2d4ab7f27">


i.e attributes and resource attributes should be at the end

mustajab-ikram commented:
> i.e attributes and resource attributes should be at the end

This was deliberately done, palash & I discussed on it:
![image](https://github.com/SigNoz/signoz/assets/94978425/b66088bf-146a-4da6-b728-8080ed923c99)
